src/WpfMath/Atoms/RowAtom.cs

```diff
@@ -43,47 +43,41 @@ public RowAtom(SourceSpan source, Atom baseAtom)
         {
         }
 
-        public RowAtom(SourceSpan source)
+        public RowAtom(SourceSpan? source)
             : base(source)
         {
             this.Elements = new List<Atom>().AsReadOnly();
         }
 
-        private RowAtom(SourceSpan source, DummyAtom previousAtom, ReadOnlyCollection<Atom> elements)
+        private RowAtom(SourceSpan? source, DummyAtom? previousAtom, ReadOnlyCollection<Atom> elements)
             : base(source)
         {
             this.PreviousAtom = previousAtom;
             this.Elements = elements;
         }
 
-        internal RowAtom(SourceSpan source, IEnumerable<Atom> elements)
+        internal RowAtom(SourceSpan? source, IEnumerable<Atom> elements)
             : base(source) =>
-            this.Elements = elements.Where(x => x != null).ToList().AsReadOnly();
-            // TODO[F]: Fix this with C# 8 migration: there shouldn't be nullable atoms in this collection
+            this.Elements = elements.ToList().AsReadOnly();
```

Comment:
You were right, change of these lines lead to test failure. Could you roll them back for now, so we'll figure it out later?